.github/workflows/turing.yaml

```diff
@@ -39,8 +39,8 @@ on:
 
 env:
   ARTIFACT_RETENTION_DAYS: 7
-  GO_VERSION: "1.20"
-  GO_LINT_VERSION: v1.51.2
+  GO_VERSION: "1.22"
+  GO_LINT_VERSION: v1.56.2
```

Comment:
Similar to what was explained in this comment [here](https://github.com/caraml-dev/merlin/pull/551/files#r1522415541), we need to use newer versions of the Go linter that actually supports Go 1.22, otherwise we'd run in OOM issues.